### ISSUE TITLE
Auto magically append / insert measures at end of score in Note input mode

### ISSUE THREAD

**Issue by sammik** [musescore/MuseScore#22600]
### Your idea

When writing music, and goes to the end of the score, new measure should appear.

### Problem to be solved

Now, when inputing notes and goes to the end of score, last note is changed, and no more measures appeares.

https://github.com/musescore/MuseScore/assets/1646034/21ae907a-6f4c-42e6-a898-956b6cf933cf

**Comment by XiaoMigros** [comment]
Maybe the measure should not automatically appear, but rather the cursor move past the last barline? That way, if another note is inputted the measure will be added, but if not, it will stay as is

**Comment by DaddyLudwig** [comment]
This would be a useful feature for me so that I could at the very least, finish a slurred phrase before I go Add -> Measures -> Insert at end of score and add another 100 or 200 measures.

**Comment by cbjeukendrup** [comment]
I agree that MuseScore should not "proactively" add measures when the user comes close to the end of the score, because that is not always desired.
Instead, the solution would indeed be that the cursor should be able to get into a "beyond the last barline" state, after which a next note being entered will cause a new measure to be added.
Visually, I've always imagined something like this:
<img width="1134" alt="Scherm­afbeelding 2024-04-27 om 18 05 09" src="https://github.com/musescore/MuseScore/assets/48658420/ae5c66a9-d44e-416c-a590-26072b781311">

**Comment by TeeDeeY** [comment]
A short term workaround could be if control-b (add measure to end) added a measure and didn’t change the current insert point / current position in score / current state (in note input mode or not.  It bugs me now when I accidentally press control-b that I am taken away from where I was working and moved to the end.  Also, sometimes I want to add a few measures to the end quickly, and control-b would be convenient.  But, I don’t want to the current location/state to change immediate.

**Comment by Ash-86** [comment]
Hi everyone, check out this related issue #24787 where similar arguments about "r" and "t" actions not automagically adding a new measure came up also.  Same Problem exists when inputting notes via midi, you really have to break your writing flow and add a measure; even if by ctrl+b which changes the cursor position thus disallowing for a continuous writing experience.

**Comment by Rusca8** [comment]
It could also be added as a toggle, so that it doesn't bother the general public but people used to it (e.g. we old finale users) could activate it.
